#24737 Extend CLI command to validate unmatched arguments

jgambarios opened the pull request:
Introduced checks for unmatched command-line arguments across all CLI commands. This enhancement prevents silent failures when an unrecognized option is provided. The changes throw an `UnmatchedArgumentException `whenever any unmatched arguments are found, signalling the user about the issue. FilesPull, FilesTree and WorkspaceMixin were modified to correct CLI parameters' behaviour and improve their descriptions.

jgambarios committed:
#24737 Extend CLI command to validate unmatched arguments

Introduced checks for unmatched command-line arguments across all CLI commands.
This enhancement prevents silent failures when an unrecognized option is provided.
The changes throw an UnmatchedArgumentException whenever any unmatched arguments are found, signalling the user about the issue.
FilesPull, FilesTree and WorkspaceMixin were modified to correct CLI parameters' behaviour and improve their descriptions.

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-24737-dotCMS-CLI-Implement-pull-command

nollymar commented on tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/files/FilesPullCommandIntegrationTest.java:
```diff
@@ -144,121 +146,7 @@ void Test_Command_Files_Pull_Option_Recursive2() throws IOException {
-    }
-
-    @Test
-    void Test_Command_Files_Pull_Option_Recursive3() throws IOException {
```
why aren't these tests required anymore?

jgambarios replied:
On those tests I was testing the command didn't fail with the use of boolean after some option flags (`--some-option true` / `--some-option false`, with the improvements on the pull command flags and the validation of the proper use of the command using the `throwIfUnmatchedArguments` those boolean after the option are no longer valid.

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-24737-dotCMS-CLI-Implement-pull-command

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-24737-dotCMS-CLI-Implement-pull-command

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-24737-dotCMS-CLI-Implement-pull-command

jgambarios committed:
#24737 "Change '--interactive=false' option to '--non-interactive'

The change replaces the `--interactive=false` command flag with the more straightforward `--non-interactive`. This choice of syntax negates the need for users to specify a 'false' value. Updates have been applied to the ContentTypeCommandIntegrationTest, README, InteractiveOptionMixin, and SiteCommandIntegrationTest files to reflect this command modification."

jgambarios committed:
#24737 Fixing integration tests

jgambarios committed:
#24737 Fixing integration tests

jgambarios committed:
#24737 Fixing integration tests

jgambarios committed:
#24737 Fixing integration tests

jgambarios committed:
#24737 Fixing integration tests
